translations/en/plugin-handbook.md

```diff
@@ -2190,14 +2190,104 @@ class Foo {
 }
 ```
 
-## <a id="toc-unit-test-your-plugin"></a> Unit Test your plugin
+## <a id="toc-unit-testing"></a>Unit Testing
 
-When developing your plugin, eventually you'll want to test it!
+There are two primary ways to test babel plugins. The first snapshot tests,
+and the other is testing the AST returned by the plugin.
+We'll use jest for this example because it supports
+snapshot testing out of the box. The example we're creating here is hosted in
+[this repo](https://github.com/brigand/babel-plugin-testing-example).
 
-The [`generator-babel-plugin`](https://github.com/babel/generator-babel-plugin) package
-sets up your tests automatically. You may also want to look at
-[`babel-plugin-tester`](https://github.com/kentcdodds/babel-plugin-tester) which
-can also make testing plugins easier.
+First we need a babel plugin, we'll put this in src/index.js.
+
+```js
+
+module.exports = function testPlugin(babel) {
+  return {
+    visitor: {
+      Identifier(path) {
+        if (path.node.name === 'foo') {
+          path.node.name = 'bar';
+        }
+      }
+    }
+  };
+};
+```
+
+### Snapshot Tests
+
+Next, install our dependencies with `npm install --save-dev babel-core jest`, and
+then we can begin writing our first test: the snapshot. Snapshot tests allow us
+to visually inspect the output of our babel plugin. We give it an input, tell it to make
+a snapshot, and it saves it to a file. We check in the snapshots into git. This allows
+us to see when we've affected the output of any of our test cases. It also gives use a diff
+in pull requests. Of course you could do this with any test framework, but with jest updating
+the snapshots is as easy as `jest -u`.
+
+```js
+// src/__tests__/index-test.js
+const babel = require('babel-core');
+const plugin = require('../');
+
+var example = `
+var foo = 1;
+if (foo) console.log(foo);
+`;
+
+it('works', () => {
+  const {code} = babel.transform(example, {plugins: [plugin]});
+  expect(code).toMatchSnapshot();
+});
+```
+
+This gives us a snapshot file in `src/__tests__/__snapshots__/index-test.js.snap`.
+
+```js
+exports[`test works 1`] = `
+"
+var bar = 1;
+if (bar) console.log(bar);"
+`;
+```
+
+If we change 'bar' to 'baz' in our plugin and run jest again, we get this:
+
+```diff
+Received value does not match stored snapshot 1.
+
+    - Snapshot
+    + Received
+
+    @@ -1,3 +1,3 @@
+     "
+    -var bar = 1;
+    -if (bar) console.log(bar);"
+    +var baz = 1;
+    +if (baz) console.log(baz);"
+```
+
+We see how our change to the plugin code affected the output of our plugin, and
+if the output looks good to us, we can run `jest -u` to update the snapshot.
+
+### AST Tests
```

Comment:
I [tried ast-pretty-print](https://github.com/brigand/babel-plugin-testing-example/blob/pretty-print/src/__tests__/__snapshots__/index-test.js.snap) and it produced a ton of output for a small piece of code. Looking at the output, I have no idea if the plugin is working correctly or not, and there's a lot of noise.

I don't see it being particularly useful for unit testing, but it could be helpful for debugging. I'd like to omit it for now unless more people want it added.